app/assets/src/components/views/discovery/DiscoverySidebar.jsx

```diff
@@ -0,0 +1,209 @@
+
+export default class DiscoverySidebar extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      stats: {
+        samples: 0,
+        projects: 0
+        // TODO (gdingle):
+        // avg_reads_per_sample: 0,
+      },
+      metadata: {
+        host: {},
+        tissue: {},
+        created_at: {}
+        // TODO (gdingle):
+        // location: {},
+      },
+      _computed: null
+    };
+  }
+
+  async componentDidMount() {
+    // TODO (gdingle): why not include genome name in payload from getSamples?
+    const genomes = keyBy("id", await getAllHostGenomes());
+    this.setState({ genomes });
+  }
+
+  static getDerivedStateFromProps(newProps, prevState) {
+    const { currentTab, projects } = newProps;
+
+    if (currentTab == "samples") {
+      const samples = DiscoverySidebar.processSamples(
+        newProps.samples,
+        prevState.genomes
+      );
+      if (!samples || !samples.length) {
+        return prevState;
+      }
+      return {
+        stats: {
+          samples: samples.length,
+          projects: uniqBy("project_id", samples).length
+          // TODO (gdingle): reads not in samples data yet
+          // avg_reads_per_sample: 0,
+        },
+        metadata: {
+          host: countBy("host_genome", samples),
+          tissue: countBy("sample_tissue", samples),
+          created_at: countBy("created_at", samples)
+          // TODO (gdingle): location not in samples data yet
+          // location: {},
+        },
+        _computed: currentTab
+      };
+    } else if (currentTab == "projects") {
+      if (!projects || !projects.length) {
+        return prevState;
+      }
+
+      const hosts = flatten(map("hosts", projects));
+      const tissues = flatten(map("tissues", projects));
+
+      const created_ats = map(
+        p => moment(p.create_at).format("YYYY-MM-DD"),
+        projects
+      );
+
+      return {
+        z: {
+          samples: sumBy("number_of_samples", projects),
+          projects: projects.length
+          // TODO (gdingle): reads not in projects data yet
+          // avg_reads_per_sample: 0,
+        },
+        metadata: {
+          // TODO (gdingle): these freq counts per project, not per sample
+          host: countBy(_ => _, hosts),
+          tissue: countBy(_ => _, tissues),
+          created_at: countBy(_ => _, created_ats)
+          // TODO (gdingle): location not in projects data yet
+          // location: {},
+        },
+        _computed: currentTab
+      };
+    } else {
+      // eslint-disable-next-line no-console
+      console.error("Not supported: " + currentTab);
+      return prevState;
+    }
+  }
+
+  static processSamples(newSamples, genomes) {
+    const samples = newSamples.map(sample => {
+      const genome = genomes[sample.host_genome_id];
+      // TODO (gdingle): best description for blanks?
+      sample.host_genome = genome ? genome.name : "unknown";
```

Comment:
Host genome should always be known but for others like sample_tissue I'd be fine with empty string

Reply:
Yeah host genome is always present, so maybe a bug to be worked out?

Reply:
I actually didn't see any unknown genomes so no worries. 

@jshoe empty string doesn't work here because we need a value to click on. The root cause of empty strings should be addressed somewhere and if "unknown" is a valid state then we should require it on db write.